cpu/nrf5x_common/clock.c

```diff
@@ -78,10 +78,18 @@ void clock_hfxo_release(void)
     irq_restore(state);
 }
 
+/**
+ * True if the low frequency clock (LFCLK) has been started by RIOT.
+ * We don't rely on NRF_CLOCK->LFCLKSTAT since that register appears to be latched
+ * for a short amount of time after a soft reset on at least nRF52832 and nRF52840.
+ * @see https://devzone.nordicsemi.com/f/nordic-q-a/35792/nrf_clock--lfclkstat-register-contents-are-not-properly-evaluated-after-a-system-reset-if-rtc-compare-event-1-or-2-are-used/138995
+ */
+static bool clock_lf_running = false;
+
 void clock_start_lf(void)
 {
     /* abort if LF clock is already running */
-    if (NRF_CLOCK->LFCLKSTAT & CLOCK_LFCLKSTAT_STATE_Msk) {
+    if (clock_lf_running) {
```

Comment:
Is there a case where this function would be called more than once anyway?
I think we could easily drop that check.

It's only called in `rtt_init()` which is run once, the call in `nimble_riot.c` could be guarded with a `if (!IS_USED(MODULE_PERIPH_RTT))`

Reply:
Moving this check to a guard in `nimble_riot.c` moves logic to somewhere where it doesn't belong to. And the product specification explicitly mentions that changing clock parameters while the clock is running is not allowed.